tidy3d/components/spice/sources/dc.py

```diff
@@ -73,6 +75,20 @@ def check_voltage(cls, val):
                 raise ValueError(f"Voltages must be finite. Currently  voltage={val}.")
         return val
 
+    @pd.validator("voltage")
+    def check_repeated_voltage(cls, val):
+        """Warn if repeated voltage values are present, treating 0 and -0 as the same value."""
+        # Normalize all zero values (both 0.0 and -0.0) to 0.0 so they are treated as duplicates
+        normalized = np.where(np.isclose(val, 0, atol=1e-10), 0.0, val)
+        unique_values = np.unique(normalized)
+        if len(unique_values) < len(val):
+            log.warning(
+                "Duplicate voltage values detected in 'voltage' array. "
+                f"Found {len(val)} values but only {len(unique_values)} are unique. "
+                "Note: '0' and '-0' are considered the same value."
+            )
```

Comment:
Generally looks good, but the one thing that bugs me a little is there's no guaranteed correspondence between what's validated here and what values the solver will compute at. Would be nice to have a single source of truth for the "computed voltages", which also e.g. defines the tolerance within which two values are considered identical.